modules/dynamic-apiserver/pkg/app/service_archive.go

```diff
@@ -0,0 +1,39 @@
+package app
+
+import (
+	"fmt"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/kuberlogic/kuberlogic/modules/dynamic-apiserver/pkg/generated/models"
+	apiService "github.com/kuberlogic/kuberlogic/modules/dynamic-apiserver/pkg/generated/restapi/operations/service"
+	"github.com/kuberlogic/kuberlogic/modules/dynamic-apiserver/pkg/util"
+	kuberlogiccomv1alpha1 "github.com/kuberlogic/kuberlogic/modules/dynamic-operator/api/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (srv *Service) ServiceArchiveHandler(params apiService.ServiceArchiveParams, _ *models.Principal) middleware.Responder {
+	ctx := params.HTTPRequest.Context()
+	// Check if service exists first
+	r := new(kuberlogiccomv1alpha1.KuberLogicService)
+	err := srv.kuberlogicClient.Get().
+		Resource(serviceK8sResource).
+		Name(params.ServiceID).
+		Do(ctx).
+		Into(r)
+	if err != nil && util.CheckStatus(err, v1.StatusReasonNotFound) {
+		msg := fmt.Sprintf("kuberlogic service not found: %s", params.ServiceID)
+		srv.log.Warnw(msg, "error", err)
+		return apiService.NewServiceArchiveNotFound().WithPayload(&models.Error{
+			Message: msg,
+		})
+	} else if err != nil {
+		msg := "error finding service"
+		srv.log.Errorw(msg, "error", err)
+		return apiService.NewServiceArchiveServiceUnavailable().WithPayload(&models.Error{
+			Message: msg,
+		})
+	}
+	// Get service archive in background
+	go srv.ArchiveKuberlogicService(&params.ServiceID)
```

Comment:
I suggest to use parameter "no-wait: true", by default it false.

Reply:
Also it can be failed if the backups can not created, but function returned 200.

Reply:
Resolved in PM: we decide to stay without wait mode